lib/promscrape/discovery/consulagent/service_node.go

```diff
@@ -0,0 +1,83 @@
+package consulagent
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/consul"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promutils"
+)
+
+// getServiceNodesLabels returns labels for Consul service nodes with given cfg.
+func getServiceNodesLabels(cfg *apiConfig) []*promutils.Labels {
+	sns := cfg.consulWatcher.getServiceNodesSnapshot()
+	var ms []*promutils.Labels
+	for svc, sn := range sns {
+		for i := range sn {
+			ms = appendTargetLabels(sn[i], ms, svc, cfg.tagSeparator, cfg.agent)
+		}
+	}
+	return ms
+}
+
+func appendTargetLabels(sn consul.ServiceNode, ms []*promutils.Labels, serviceName, tagSeparator string, agent *consul.Agent) []*promutils.Labels {
+	const metaPrefix = "__meta_consulagent_"
+
+	var addr string
+	if sn.Service.Address != "" {
```

Comment:
Could you please add a comment why `sn.Service.Address` can be empty?
It will be easier to understand later why this handling was added.

Reply:
done